subsys/bluetooth/host/classic/sdp.c

```diff
@@ -1705,6 +1705,10 @@ static void sdp_client_params_iterator(struct bt_sdp_client *session)
 		sys_slist_remove(&session->reqs, NULL, &param->_node);
 		/* Invalidate cached param in context */
 		session->param = NULL;
+		if (session->rec_buf != NULL) {
```

Comment:
current session rec_buf  would be alloc  when sdp_client_connected. if do alloc rec_buf when user request every discovery. then rec_buf  maybe need to delete within sdp_client_connected.

However I  am some  doubts.
recv_len would be reset when discovery finished. why rec_buf  would be  some garbage data？